src/hooks/protections/received_event.cpp

```diff
@@ -741,6 +741,38 @@ namespace big
 		default: break;
 		}
 
+		if (event_id == 74) // NETWORK_PTFX (eNetworkEvents::NETWORK_PTFX is 75 so this would never trigger above)
+		{
+			if (!plyr)
+			{
+				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+				return;
+			}
+			if (plyr->block_ptfx)
+			{
+				g_pointers->m_gta.m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
+				return;
+			}
+			auto now = time(0);
+			if (difftime(now,plyr->ptfx_lastsend)>1)
+			{
+				plyr->ptfx_sent = 0;
+				plyr->ptfxSpamNotificationSent = false; // Reset as they have stopped spamming
+			}
+			plyr->ptfx_lastsend = now;
+			plyr->ptfx_sent += 1;
+			if (plyr->ptfx_sent > 3) // Spamming ptfx

```

Comment:
No need to do all this logic, there's a ratelimit utility object that keeps track of all these things.
See the implementation [here](https://github.com/YimMenu/YimMenu/blob/master/src/services/players/rate_limiter.hpp).

Here are already a few implemented on the player object:
https://github.com/YimMenu/YimMenu/blob/cfb011d192c487cb0e43c66d7530f991b824b2bb/src/services/players/player.hpp#L71-L74